x-pack/plugin/runtime-fields/src/test/java/org/elasticsearch/xpack/runtimefields/mapper/ScriptFieldMapperTests.java

```diff
@@ -106,46 +107,32 @@ public void testStoredScriptsAreNotSupported() throws Exception {
     }
 
     public void testUnsupportedRuntimeType() throws Exception {
-        XContentBuilder mapping = XContentFactory.jsonBuilder()
-            .startObject()
-            .startObject("_doc")
-            .startObject("properties")
-            .startObject("field")
-            .field("type", "script")
-            .field("runtime_type", "unsupported")
-            .startObject("script")
-            .field("source", "keyword('test')")
-            .field("lang", "test")
-            .endObject()
-            .endObject()
-            .endObject()
-            .endObject()
-            .endObject();
-
-        MapperParsingException exc = expectThrows(MapperParsingException.class, () -> createIndex("test", Settings.EMPTY, mapping));
+        MapperParsingException exc = expectThrows(
+            MapperParsingException.class,
+            () -> createIndex("test", Settings.EMPTY, mapping("unsupported"))
+        );
         assertEquals("Failed to parse mapping: runtime_type [unsupported] not supported", exc.getMessage());
     }
 
+    public void testKeyword() throws IOException {
+        MapperService mapperService = createIndex("test", Settings.EMPTY, mapping("keyword")).mapperService();
+        FieldMapper mapper = (FieldMapper) mapperService.documentMapper().mappers().getMapper("field");
+        assertThat(mapper, instanceOf(ScriptFieldMapper.class));
+        assertEquals(Strings.toString(mapping("keyword")), Strings.toString(mapperService.documentMapper()));
+    }
+
+    public void testLong() throws IOException {
+        MapperService mapperService = createIndex("test", Settings.EMPTY, mapping("long")).mapperService();
+        FieldMapper mapper = (FieldMapper) mapperService.documentMapper().mappers().getMapper("field");
+        assertThat(mapper, instanceOf(ScriptFieldMapper.class));
+        assertEquals(Strings.toString(mapping("long")), Strings.toString(mapperService.documentMapper()));
+    }
```

Comment:
maybe adding one of these methods could even be automatic once a new supported field type is added to the map? not sure that it's important now though

Reply:
Its a good idea. We could loop through the list but I kind of like having a method for each one.